### PR TITLE
add unit test for func parsePorts and validate

### PR DESCRIPTION
**Pull request by hanxiaoshuai**
**What this PR does / why we need it**:
add unit test for func parsePorts and validate
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/service_basic_test.go [hunk 1]
```diff
@@ -152,3 +153,174 @@ func TestServiceBasicGenerate(t *testing.T) {
 		}
 	}
 }
+
+func TestParsePorts(t *testing.T) {
+	tests := []struct {
+		portstring       string
+		expectport       int32
+		expecttargetport intstr.IntOrString
+		expecterr        string
```
Comment (php-coder) [inline pkg/kubectl/service_basic_test.go]: Consider using camleCase naming.

File: pkg/kubectl/service_basic_test.go [hunk 2]
```diff
@@ -152,3 +153,174 @@ func TestServiceBasicGenerate(t *testing.T) {
...
+	}{
+		{
+			portstring:       "3232",
+			expectport:       3232,
+			expecttargetport: intstr.IntOrString{Type: intstr.Int, IntVal: 3232},
+			expecterr:        "",
```
Comment (php-coder) [inline pkg/kubectl/service_basic_test.go]: Perhaps, you can omit it at all (`expecterr` will be `nil` in that case and should work the same as an empty string).

File: pkg/kubectl/service_basic_test.go (hunk 1 above)
Comment (hanxiaoshuai) [reply pkg/kubectl/service_basic_test.go]: Done

File: pkg/kubectl/service_basic_test.go (hunk 2 above)
Comment (hanxiaoshuai) [reply pkg/kubectl/service_basic_test.go]: @php-coder Done ,Thanks for reviewing

**Review by janetkuo — APPROVED** [review]

**Pull request merged**